### PR TITLE
Json/valid cols

### PR DESCRIPTION
**Pull request by Koustavd18**
Fix: This PR changes the JSON view to show the keys with a valid value

### CODE REVIEW

**Review by balaji-jr — CHANGES_REQUESTED** [review]
To find the valid headers you are iterating the log object twice which is unnecessary. 
And not clear why this is needed and why its included in the finding valid headers logic ?
`log[header] === null ? 'null'`

File: src/pages/Stream/Views/Explore/JSONView.tsx [hunk 1]
```diff
@@ -67,19 +70,25 @@ const Row = (props: {
 	shouldHighlight: (val: number | string | Date | null) => boolean;
 }) => {
 	const { log, headers, disableHighlight, shouldHighlight } = props;
+	const validHeaders = headers.filter((header) => {
```
Comment (balaji-jr) [inline src/pages/Stream/Views/Explore/JSONView.tsx:73]: Why this iteration is necessary ?

**Review by balaji-jr — CHANGES_REQUESTED** [review]

File: src/pages/Stream/Views/Explore/JSONView.tsx [hunk 2]
```diff
@@ -72,14 +72,18 @@ const Row = (props: {
...
-					_.map(headers, (header, index) => (
-						<Item
-							header={header}
-							key={index}
-							value={_.toString(log[header])}
-							highlight={disableHighlight ? false : shouldHighlight(log[header])}
-						/>
-					))
+					_.map(headers, (header, index) => {
+						if (!_.toString(log[header])) return;
```
Comment (balaji-jr) [inline src/pages/Stream/Views/Explore/JSONView.tsx:76]: Why not iterate over key-value pairs then?

**Review by balaji-jr — CHANGES_REQUESTED** [review]

File: src/pages/Stream/Views/Explore/JSONView.tsx [hunk 3]
```diff
@@ -61,25 +61,28 @@ const CopyIcon = (props: { log: Log }) => {
...
-					_.map(headers, (header, index) => (
-						<Item
-							header={header}
-							key={index}
-							value={_.toString(log[header])}
-							highlight={disableHighlight ? false : shouldHighlight(log[header])}
-						/>
-					))
+					_.map(log, (value, key) => {
+						//skiping fields with empty strings
```
Comment (balaji-jr) [inline src/pages/Stream/Views/Explore/JSONView.tsx:75]: Not sure why this is still needed.
If you are iterating over a log object, why you'd want to test whether its empty or not ?

File: src/pages/Stream/Views/Explore/JSONView.tsx [hunk 4]
```diff
@@ -61,25 +61,28 @@ const CopyIcon = (props: { log: Log }) => {
...
+						if (!_.toString(value)) return;
```
Comment (balaji-jr) [inline src/pages/Stream/Views/Explore/JSONView.tsx:76]: empty line after exit condition.

**Review by balaji-jr — CHANGES_REQUESTED** [review]

**Comment by balaji-jr** [comment]
I will address the remaining changes on https://github.com/parseablehq/console/pull/303

**Review by balaji-jr — DISMISSED** [review]

**Review by nikhilsinhaparseable — APPROVED** [review]
looks good

**Pull request merged**